src/mscorlib/shared/System/Buffers/OwnedMemory.cs

```diff
@@ -7,46 +7,88 @@
 
 namespace System.Buffers
 {
+    /// <summary>
+    /// Owner of Memory<typeparamref name="T"/> that provides appropriate lifetime management mechanisms for it.
+    /// </summary>
     public abstract class OwnedMemory<T> : IDisposable, IRetainable
     {
+        /// <summary>
+        /// The number of items in the Memory<typeparamref name="T"/>.
+        /// </summary>
         public abstract int Length { get; }
 
+        /// <summary>
+        /// Returns a span wrapping the underlying memory.
+        /// </summary>
         public abstract Span<T> Span { get; }
 
+        /// <summary>
+        /// Returns a Memory<typeparamref name="T"/> if the underlying memory has not been freed.
+        /// </summary>
+        /// <exception cref="System.ObjectDisposedException">
+        /// Thrown when the underlying memory has already been disposed.
+        /// </exception>
         public Memory<T> Memory
         {
-            get 
+            get
             {
-                if (IsDisposed) 
+                if (IsDisposed)
                 {
-                    ThrowHelper.ThrowObjectDisposedException(nameof(OwnedMemory<T>), ExceptionResource.Memory_ThrowIfDisposed);
+                    ThrowHelper.ThrowObjectDisposedException_MemoryDisposed(nameof(OwnedMemory<T>));
```

Comment:
Bit odd to have a unique exception handler; then add a chunky `nameof(OwnedMemory<T>)` param?

Reply:
In general, I think we'll need to do some clean up for the exceptions and using ThrowHelper. I hope at some point we'll have a shared ThrowHelper repos too and revisit what parameters every helper method is using.

Reply:
> we'll have a shared ThrowHelper across repos too

The eventual solution should be to have IL rewriter or some similar tech to do this optimization for you. ThrowHelper is a pain to maintain. Until we have automated way to do this, we want to use only in the places that really matter. We do not want to share accross repos.

Reply:
@tarekgh I think @benaadams has valid point. We should move the chunky name into the throw helper and not pass it as an argument. You can append a commit to this PR to fix it - the mirror will then mirror it back to CoreFX.